src/routes/safe/container/index.tsx

```diff
@@ -29,16 +29,17 @@ const AddressBookTable = lazy(() => import('src/routes/safe/components/AddressBo
 
 const Container = (): React.ReactElement => {
   const featuresEnabled = useSelector(currentSafeFeaturesEnabled)
-  const owners = useSelector(currentSafeOwners)
+  const { address, owners } = useSelector(currentSafeWithNames)
   const isSafeLoaded = owners.length > 0
   const [hasLoadFailed, setHasLoadFailed] = useState<boolean>(false)
 
   const addressFromUrl = extractSafeAddress()
-  useLoadSafe(addressFromUrl) // load initially
-  useSafeScheduledUpdates(addressFromUrl) // load every X seconds
+  useLoadSafe(address) // load initially
+  useSafeScheduledUpdates(address) // load every X seconds
 
   useEffect(() => {
-    if (isSafeLoaded) {
+    if (isSafeLoaded && address === addressFromUrl) {
```

Comment:
I am not sure if we need this comparison here. Just using the address from redux instead of the URL adds the necessary rerender already to refresh the load state on poll.